### PR TITLE
Support scheduled events

### PR DESCRIPTION
**Pull request by johnyherangi**
## Description

Add support for scheduled CloudWatch events.
https://docs.aws.amazon.com/lambda/latest/dg/services-cloudwatchevents.html

>Note: Currently not compatible with Serverless schedules as they are not configured to send the whole event (only the input)

## Example

```ts
@Controller()
export class Controller {
    @Schedule("arn:aws:events:us-east-2:123456789012:rule/my-schedule")
    public async schedule(event: ScheduledEvent) {
        // ...
    }
}
```

### CODE REVIEW

File: src/Router.ts [hunk 1]
```diff
@@ -41,6 +41,13 @@ class Router {
      */
     public route(event: SQSEvent, context: Context): Promise<void>
 
+    /**
+     *
+     * @param event
+     * @param context
+     */
```
Comment (jackholmes1010) [inline src/Router.ts]: Can we fill this in pls

File: src/typeGuards.ts [hunk 2]
```diff
@@ -14,3 +14,9 @@ export const isSqsEvent = (event: unknown): event is SQSEvent => {
         undefined
     )
 }
+
+export const isScheduledEvent = (event: unknown): event is ScheduledEvent => {
+    const e = event as ScheduledEvent
+    // https://docs.aws.amazon.com/lambda/latest/dg/services-cloudwatchevents.html
+    return e?.["detail-type"].toLowerCase() === "scheduled event"
```
Comment (jackholmes1010) [inline src/typeGuards.ts]: It's probably not necessary, but should we add `?.toLowerCase()`, just to be more safe?

Comment (johnyherangi) [reply src/typeGuards.ts]: yep, can't hurt

**Review by jackholmes1010 — APPROVED** [review]
Nice, looks good.

**Review by johnmikk — APPROVED** [review]
Nice!

**Pull request merged**